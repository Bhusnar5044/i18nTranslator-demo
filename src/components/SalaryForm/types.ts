export type Props = {
    // SalaryForm prop type
    click: () => void;
};
