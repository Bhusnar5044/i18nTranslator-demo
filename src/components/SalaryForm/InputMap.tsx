import { Box, Button, Textfield } from '@Lib';
import React, { useState } from 'react';

interface Node {
    name: string;
    value: number;
    children?: Node[];
}

interface Props {
    node: Node;
    onUpdateNode: (node: Node) => void;
    nested?: boolean;
}

const NodeTextInput: React.FC<Props> = ({ node, onUpdateNode, nested }) => {
    const [localNode, setLocalNode] = useState(node);

    const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setLocalNode((prev) => ({ ...prev, name: e.target.value }));
    };

    const handleValueChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setLocalNode((prev) => ({ ...prev, value: parseInt(e.target.value) }));
    };

    const handleChildUpdate = (childIndex: number, updatedChild: Node) => {
        setLocalNode((prev) => {
            const updatedChildren = [...prev.children!];
            updatedChildren[childIndex] = updatedChild;
            return { ...prev, children: updatedChildren };
        });
    };

    const handleAddChild = () => {
        setLocalNode((prev) => ({
            ...prev,
            children: [...prev.children!, { name: '', value: 0 }],
        }));
    };

    const handleRemoveChild = (childIndex: number) => {
        setLocalNode((prev) => {
            const updatedChildren = [...prev.children!];
            updatedChildren.splice(childIndex, 1);
            return { ...prev, children: updatedChildren };
        });
    };

    const handleSave = () => {
        onUpdateNode(localNode);
    };

    return (
        <Box display="flex" flexDirection="row" flexWrap="wrap" gap="1rem">
            <Box display="flex" flexDirection="row" gap="1rem">
                <Textfield label="name" value={localNode.name} onChange={handleNameChange} />
                <Textfield
                    //  type="number"
                    label="Value"
                    value={localNode.value}
                    onChange={handleValueChange}
                />
            </Box>
            {localNode.children?.map((child, index) => (
                <Box display="flex" flexDirection="column" gap="3rem" key={index}>
                    <NodeTextInput
                        node={child}
                        onUpdateNode={(updatedChild) => handleChildUpdate(index, updatedChild)}
                        nested
                    />
                    {/* <button onClick={() => handleRemoveChild(index)}>Remove</button>
                    <br /> */}
                </Box>
            ))}
            {/* <button onClick={handleAddChild}>Add Child</button> */}
            {!nested && <Button onClick={handleSave}>Save</Button>}
        </Box>
    );
};

export default NodeTextInput;
