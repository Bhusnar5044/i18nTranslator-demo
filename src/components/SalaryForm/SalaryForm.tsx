import { Box, Text } from '@Lib';
import ContainerLoader from '@components/ContainerLoader';
import { AppState, useAppSelector } from '@store';
import { editSalary, getSalaryRequest } from '@store/salary';
import { FC, memo, useCallback, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import NestedTextInput from './NestedTextInput/NestedTextInput';

export const SalaryForm: FC = memo(() => {
    const dispatch = useDispatch();
    const { isLoading, expenditure } = useAppSelector((state: AppState) => state.salary);
    const onSubmit = useCallback(
        (node: any) => {
            console.log({ node });
            dispatch(editSalary(node));
        },
        [dispatch],
    );

    useEffect(() => {
        dispatch(getSalaryRequest());
    }, [dispatch]);

    return (
        <Box flexDirection="column" gap="3rem">
            <Text textVariant="h3">Expenditure Details</Text>
            {isLoading && <ContainerLoader />}
            {!isLoading && <NestedTextInput node={expenditure} onSubmit={onSubmit} />}
        </Box>
    );
});

SalaryForm.displayName = 'SalaryForm';
