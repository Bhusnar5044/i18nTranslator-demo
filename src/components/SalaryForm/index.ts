export { SalaryForm as default } from './SalaryForm';
