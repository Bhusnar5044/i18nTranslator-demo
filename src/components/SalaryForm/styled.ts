import { styled } from '@utils';

export const Form = styled.form`
    display: flex;
    flex-direction: column;
    gap: 1rem;
`;
