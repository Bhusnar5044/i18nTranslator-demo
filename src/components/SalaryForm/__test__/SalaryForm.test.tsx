import { renderWithStore } from '@test-utils';
import { SalaryForm } from '../SalaryForm';

describe('SalaryForm', () => {
    it('should render properly', () => {
        const { container } = renderWithStore(<SalaryForm />);
        expect(container).toMatchSnapshot();
    });
});
