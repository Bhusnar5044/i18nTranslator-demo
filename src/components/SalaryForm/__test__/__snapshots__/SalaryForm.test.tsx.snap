// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalaryForm should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: 3rem;
}

.c2 {
  display: flex;
  flex-direction: column;
  align-items: start;
  gap: 2rem;
}

.c3 {
  display: flex;
  flex-wrap: wrap;
  gap: 1rem;
}

.c7 {
  display: inline-block;
  outline: 0;
  cursor: pointer;
  text-align: center;
  border: 0;
  padding: 7px 16px;
  min-height: 36px;
  min-width: 36px;
  color: #ffffff;
  background: #008060;
  border-radius: 4px;
  font-weight: 500;
  font-size: 14px;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 0px 0px,rgba(0, 0, 0, 0.2) 0px -1px 0px 0px inset;
}

.c7 :hover {
  background: #006e52;
}

.c1 {
  margin: 0;
  color: inherit;
  font-size: 2.4rem;
  font-weight: 400;
  letter-spacing: -0.04rem;
  line-height: 3.2rem;
  text-align: initial;
}

.c4 {
  position: relative;
}

.c6 {
  position: absolute;
  font-size: 1rem;
  left: 0;
  top: 50%;
  transform: translateY(-50%);
  background-color: white;
  color: gray;
  padding: 0 0.3rem;
  margin: 0 0.5rem;
  transition: 0.1s ease-out;
  transform-origin: left top;
  pointer-events: none;
}

.c5 {
  font-size: 1rem;
  outline: none;
  border: 1px solid gray;
  border-radius: 5px;
  padding: 0.5rem 0.7rem;
  color: gray;
  transition: 0.1s ease-out;
}

.c5:focus {
  border-color: #6200ee;
}

.c5:focus+label {
  color: #6200ee;
  top: 0;
  transform: translateY(-50%) scale(0.9);
}

.c5:not(:placeholder-shown)+label {
  top: 0;
  transform: translateY(-50%) scale(0.9);
}

<div>
  <div
    class="c0"
    flexdirection="column"
    gap="3rem"
  >
    <h3
      class="c1"
      textalign="initial"
      textcolor="inherit"
      textvariant="h3"
    >
      line1
    </h3>
    <div
      alignitems="start"
      class="c2"
      flexdirection="column"
      gap="2rem"
    >
      <div
        class="c3"
        flexwrap="wrap"
        gap="1rem"
      >
        <div
          class="c4"
        >
          <input
            class="c5"
            type="text"
          />
          <label
            class="c6"
          />
        </div>
      </div>
      <button
        class="c7"
        type="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
